### PR TITLE
[DC-3094] add proxy to preference-frontend

### PR DESCRIPTION
**Pull request by lxol**

### CODE REVIEW

**Review by elabeca — APPROVED** [review]

**Review by muralishub — APPROVED** [review]

File: app/uk/gov/hmrc/channelpreferencesfrontend/config/DIModule.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import com.google.inject.AbstractModule
+import com.google.inject.name.Names.named
+import com.typesafe.config.ConfigException
+import play.api.{ Configuration, Environment }
+
+class DIModule(environment: Environment, configuration: Configuration) extends AbstractModule {
+
+  override def configure(): Unit =
+    bindString(s"preferencesFrontendUrl", "preferencesFrontendUrl")
```
Comment (elabeca) [inline app/uk/gov/hmrc/channelpreferencesfrontend/config/DIModule.scala:27]: val serviceUrl = baseUrl("preferences-frontend") by importing uk.gov.hmrc.play.bootstrap.config.ServicesConfig and having a microservice.services section. Wouldn't want the next person in line using this as an example to fetch their service config for channel-preferences configs.

Comment (elabeca) [reply app/uk/gov/hmrc/channelpreferencesfrontend/config/DIModule.scala:27]: .

**Review by elabeca — CHANGES_REQUESTED** [review]

**Review by rwalpole — APPROVED** [review]

File: app/uk/gov/hmrc/channelpreferencesfrontend/config/DIModule.scala (hunk 1 above)
Comment (elabeca) [inline app/uk/gov/hmrc/channelpreferencesfrontend/config/DIModule.scala:27]: val serviceUrl = baseUrl("preferences-frontend") by importing uk.gov.hmrc.play.bootstrap.config.ServicesConfig and having a microservice.services section. Wouldn't want the next person in line using this as an example to fetch their service config for channel-preferences configs.

Comment (elabeca) [inline app/uk/gov/hmrc/channelpreferencesfrontend/config/DIModule.scala:27]: val serviceUrl = baseUrl("preferences-frontend") by importing uk.gov.hmrc.play.bootstrap.config.ServicesConfig and having a microservice.services section. Wouldn't want the next person in line using this as an example to fetch their service config for channel-preferences configs.

**Review by elabeca — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/channelpreferencesfrontend/config/DIModule.scala (hunk 1 above)
Comment (lxol) [reply app/uk/gov/hmrc/channelpreferencesfrontend/config/DIModule.scala:27]: I  wanted not to confuse "proxying" from the normal call to microservice. It is temporary, when  preference-frontend is fully migrated to channel-preferences-frontend we remove proxy with all associated configs.

**Review by elabeca — APPROVED** [review]

**Pull request merged**